### ISSUE TITLE
Failed to import - dayjs (related to dts transformer)

### ISSUE THREAD

**Issue by echoptic** [esm-dev/esm.sh#581]
## Failing module

- **GitHub**: https://github.com/iamkun/dayjs
- **npm**: https://www.npmjs.com/package/dayjs

```ts
import dayjs from "https://esm.sh/v113/dayjs@1.11.7";
import relativeTime from "https://esm.sh/v113/dayjs@1.11.7/plugin/relativeTime.js";

dayjs.extend(relativeTime);

// @ts-ignore Deno doesn't recognize the function fromNow
const ago = dayjs(props.time * 1000).fromNow();
```

## Additional info

- **esm.sh version**: v113
- **Deno version**: 1.32.1

**Comment by ije** [comment]
thanks! continue the topic of #576, i think we need to resolve `declare module 'dayjs'` to `declare module 'https://esm.sh/v113/dayjs@1.11.7/index.d.ts'`, since deno uses URL as the module name

```ts
// esm.sh ignores below types
declare module 'https://esm.sh/v113/dayjs@1.11.7/index.d.ts' {
  interface Dayjs {
    fromNow(withoutSuffix?: boolean): string
    from(compared: ConfigType, withoutSuffix?: boolean): string
    toNow(withoutSuffix?: boolean): string
    to(compared: ConfigType, withoutSuffix?: boolean): string
  }
}
```

**Comment by echoptic** [comment]
> 

@ije You are right, just tested to renaming 'dayjs' to 'https://esm.sh/v113/dayjs@1.11.7/index.d.ts' in the `relativeTime.d.ts` to confirm it, and the autocomplete works. Can you put me in the right direction about what needs to be changed to make this work?

**Comment by echoptic** [comment]
> thanks! continue to the topic of #576, i think we need to resolve `declare module 'dayjs'` to `declare module 'https://esm.sh/v113/dayjs@1.11.7/index.d.ts'` in the `.d.ts`, since deno uses URL as the module name:
> 
> ```ts
> // esm.sh ignores below types
> declare module 'https://esm.sh/v113/dayjs@1.11.7/index.d.ts' {
>   interface Dayjs {
>     fromNow(withoutSuffix?: boolean): string
>     from(compared: ConfigType, withoutSuffix?: boolean): string
>     toNow(withoutSuffix?: boolean): string
>     to(compared: ConfigType, withoutSuffix?: boolean): string
>   }
> }
> ```

@ije You are right, just tested to renaming 'dayjs' to 'https://esm.sh/v113/dayjs@1.11.7/index.d.ts' in the relativeTime.d.ts to confirm it, and the autocomplete works. Can you put me in the right direction about what needs to be changed to make this work?

**Comment by ije** [comment]
maybe start from here:
https://github.com/esm-dev/esm.sh/blob/b058371c248e482902db12529fb9dff56f074a18/server/dts_transform.go#L104

**Comment by echoptic** [comment]
Thanks, I will check it out tomorrow.

**Comment by echoptic** [comment]
im not sure where exactly is the 'declare module "dayjs"' is being resolved. i tried just returning the string "something" under the line `if kind == "declareModule" {` and the only file that contains the string 'something' was this:
```sh
rg something vendor/
vendor/localhost_8080/v114/dayjs@1.11.7/locale/index.d.ts
3:declare module 'something' {
```

**Comment by ije** [comment]
fixed in v115

**Comment by ije** [comment]
> im not sure where exactly is the 'declare module "dayjs"' is being resolved. i tried just returning the string "something" under the line `if kind == "declareModule" {` and the only file that contains the string 'something' was this:
> 
> ```shell
> rg something vendor/
> vendor/localhost_8080/v114/dayjs@1.11.7/locale/index.d.ts
> 3:declare module 'something' {
> ```

also removed some unclear code of the dts transformer
